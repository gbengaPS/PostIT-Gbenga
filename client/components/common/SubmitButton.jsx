import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description SubmitButton component
 *
 * @param {object} props -prop object
 *
 * @returns {jsx} -jsx representation of the component
 */
const SubmitButton = props => (
  <div>
    <p className="center">
    <input
      type="submit"
      value={props.submitValue}
      className="btn light-blue darken-4 "
      disabled={props.disabled}
    />
    </p>
  </div>
);
SubmitButton.propTypes = {
  submitValue: PropTypes.string.isRequired,
  disabled: PropTypes.string,
};
export default SubmitButton;
