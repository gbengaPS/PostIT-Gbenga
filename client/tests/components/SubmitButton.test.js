import React from 'react';
import SubmitButton from '../../components/common/SubmitButton';

describe('SubmitButton', () => {
  const props = {
    submitValue: 'submit',
    disabled: ''
  };
  const SubmitButtonWrapper = shallow(<SubmitButton {...props}/>);
  it('should render correctly', () => {
    expect(SubmitButtonWrapper).toMatchSnapshot();
  });
});
