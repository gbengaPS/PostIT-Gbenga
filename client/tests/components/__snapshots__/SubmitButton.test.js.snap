// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render correctly 1`] = `
<div
  className="row"
>
  <input
    className="btn light-blue darken-4 col m4 offset-m4"
    disabled=""
    type="submit"
    value="submit"
  />
</div>
`;
